lxd/db/storage_volumes.go

```diff
@@ -72,7 +72,10 @@ WHERE storage_volumes.type = ?
 			Description: r[2].(string),
 			PoolName:    r[3].(string),
 			ProjectName: r[4].(string),
-			NodeID:      r[5].(int64),
```

Comment:
@masnax can this be in a separate PR? Also I'd like to understand more about why this change is required. Thanks

Reply:
To build the URI for storage volumes we need the node id of the storage volume. This field is nullable so either `StorageVolumeArgs.NodeID` needs to be a `*int64` or we need to default `storage_volumes.node_id` to `-1` if it's NULL.

Reply:
Can you handled that via IFNULL in the query?

Reply:
Also, but something for another day, there's no reason that field should be NULLable AFAIK.

Reply:
That's certainly possible, but I was trying to get away from using fake `nil` values like `-1`.

Reply:
what value was it before this change?

Reply:
Previously two separate queries were being performed: one for volumes with node_ids, and one without.
This is the whole view:
```sql
CREATE VIEW projects_used_by_ref (name,
    value) AS
  SELECT projects.name,
    printf('/1.0/instances/%s?project=%s',
    "instances".name,
    projects.name)
    FROM "instances" JOIN projects ON project_id=projects.id UNION
  SELECT projects.name,
    printf('/1.0/images/%s?project=%s',
    images.fingerprint,
    projects.name)
    FROM images JOIN projects ON project_id=projects.id UNION
  SELECT projects.name,
    printf('/1.0/storage-pools/%s/volumes/custom/%s?project=%s&target=%s',
    storage_pools.name,
    storage_volumes.name,
    projects.name,
    nodes.name)
    FROM storage_volumes JOIN storage_pools ON storage_pool_id=storage_pools.id JOIN nodes ON node_id=nodes.id JOIN projects ON project_id=projects.id WHERE storage_volumes.type=2 UNION
  SELECT projects.name,
    printf('/1.0/storage-pools/%s/volumes/custom/%s?project=%s',
    storage_pools.name,
    storage_volumes.name,
    projects.name)
    FROM storage_volumes JOIN storage_pools ON storage_pool_id=storage_pools.id JOIN projects ON project_id=projects.id WHERE storage_volumes.type=2 AND storage_volumes.node_id IS NULL UNION
  SELECT projects.name,
    printf('/1.0/profiles/%s?project=%s',
    profiles.name,
    projects.name)
    FROM profiles JOIN projects ON project_id=projects.id UNION
  SELECT projects.name,
    printf('/1.0/networks/%s?project=%s',
    networks.name,
    projects.name)
    FROM networks JOIN projects ON project_id=projects.id UNION
  SELECT projects.name,
    printf('/1.0/network-acls/%s?project=%s',
    networks_acls.name,
    projects.name)
    FROM networks_acls JOIN projects ON project_id=projects.id;
```

Here are the relevant parts:
```sql
SELECT projects.name,
    printf('/1.0/storage-pools/%s/volumes/custom/%s?project=%s&target=%s',
    storage_pools.name,
    storage_volumes.name,
    projects.name,
    nodes.name)
    FROM storage_volumes JOIN storage_pools ON storage_pool_id=storage_pools.id JOIN nodes ON node_id=nodes.id JOIN projects ON project_id=projects.id WHERE storage_volumes.type=2 UNION
  SELECT projects.name,
    printf('/1.0/storage-pools/%s/volumes/custom/%s?project=%s',
    storage_pools.name,
    storage_volumes.name,
    projects.name)
    FROM storage_volumes JOIN storage_pools ON storage_pool_id=storage_pools.id JOIN projects ON project_id=projects.id WHERE storage_volumes.type=2 AND storage_volumes.node_id IS NULL UNION
```

Reply:
> That's certainly possible, but I was trying to get away from using fake `nil` values like `-1`

Can you decouple this change from this PR, it would be worth discussing separately. 
When we started going down this route for filtering it made sense, as the inputs were optional.
But with this change it feels like its starting to go too far in using nils to indicate something that isn't optional, and perhaps this is a schema issue that we should either fix or try and mask as early as possible and not propagate it.

And assuming we were going to support nil here, its not been completed yet, as  the query itself is still doing `IFNULL(storage_volumes.node_id, -1)` and further down there's a reference to `v.NodeID < 0`.